arbor/mechcat.cpp

```diff
@@ -579,4 +575,28 @@ std::pair<mechanism_ptr, mechanism_overrides> mechanism_catalogue::instance_impl
 
 mechanism_catalogue::~mechanism_catalogue() = default;
 
+// Load a catalogue from a shared object
+const mechanism_catalogue& load_catalogue(const std::string& fn) {
+    typedef const void* global_catalogue_t();
+
+    auto plugin = dlopen(fn.c_str(), RTLD_LAZY);
+    if (!plugin) {
+        auto error = dlerror();
+        throw arb::dynamic_catalogue_error(fn, error);
```

Comment:
I fear this will lead to a discussion on principles, but do we target windows at all?
Regardless, his interface will not work on Windows, as `dlopen` does not exist there. 
So, when building an analogue for this functionality in Windows, we would have to use 
a different API which can construct it's exceptions differently.

Reply:
Supposing that we do support a port, the interface (i.e. our Arbor interface) can be made agnostic by removing the error text from `dlopen`/`LoadLibraryW` (or by having, ugh, a variant type for the error text). Then it comes down to some `#ifdef`s in the code and/or some wrapping of the calls.